browser-extension/common/src/popup/components/PopupTabs/index.tsx

```diff
@@ -24,6 +25,23 @@ const PopupTabs: React.FC = () => {
   const [isRuleDropdownOpen, setIsRuleDropdownOpen] = useState(false);
   const [activeTabKey, setActiveTabKey] = useState(PopupTabKey.PINNED_RULES);
 
+  useEffect(() => {
+    chrome.tabs.query({ currentWindow: true, active: true }, ([activeTab]) => {
+      chrome.runtime.sendMessage(
+        {
+          tabId: activeTab.id,
+          action: EXTENSION_MESSAGES.GET_EXECUTED_RULES,
+        },
+        (rules) => {
+          setExecutedRulesCount(rules.length);
```

Comment:
If rules is empty, is this handled inside setExecutedRulesCount ?

Reply:
Yes. Handled in the callback